src/components/CharacterTab.jsx

```diff
@@ -222,6 +233,29 @@ export default function CharacterTab() {
     cellStyle: { display: 'flex' },
   }), [])
 
+  const externalFilterChanged = useCallback(() => {
+    characterGrid.current.api.onFilterChanged()
+  }, [])
+
+  const doesExternalFilterPass = useCallback(
+    (node) => {
+      const charinfo = Utils.clone(DB.getMetadata().characters)
```

Comment:
After some profiling, I found most of the lag was coming from this.

This clone isn't necessary here since no modifications are made, and DB.getMetadata().characters is a massive json, so its triggering a lag spike by cloning it on every keystroke